mne/connectivity/cfc.py

```diff
@@ -0,0 +1,529 @@
+
+    if np.max(win) == np.min(win):
+        raise RuntimeError('win values lead to divide by 0.')
+
+    # normalize the gaussian window
+    win = (win - np.min(win)) / (np.max(win) - np.min(win))
+
+    # Construct the signal with many levels of modulation
+    pac_signal = np.zeros((n_epochs, n_times))
+
+    for sig in range(n_epochs):
+        # Generate amplitude envelope of modulated signal
+        amp_fa = (max_amp * ((1. - win[sig]) *
+                             np.sin(2 * np.pi * f_phase * times)
+                             + 1. + win[sig]) / 2.)
+        # Generate the phase amplitude coupled signal
+        pac_signal[sig] = (amp_fa * np.sin(2. * np.pi * f_amplitude * times) +
+                           max_amp * np.sin(2. * np.pi * f_phase * times) +
+                           white_noise[sig])
+
+    if len(pac_signal) != n_epochs:
+        raise RuntimeError('Data not equal to epochs.')
+
+    # mix the trials
+    order = np.argsort(rng.randn(len(pac_signal)))
+    pac_signal = pac_signal[order, :]
+
+    return pac_signal
+
+
+def cross_frequency_coupling(data, sfreq, phase_freq, n_cycles,
+                             freqs, tmin, tmax, n_jobs=1):
+    """ Compute the cross frequency coupling.
+
+    Parameters
+    ----------
+    data : array, shape (n_times,)
+        Signal time series.
+    sfreq : float
+        Sampling frequency.
+    phase_freq : float
+        The phase modulating frequency in Hz. (eg. 10.)
+    n_cycles : float | array of float
+        Number of cycles used to compute Morlet wavelets.
+    freqs : array
+        Frequency range of interest (1 x Frequencies)
+    tmin : float
+        Start time before event.
+    tmax : float
+        End time after event.
+    n_jobs : int
+        Number of jobs to run in parallel. Default 1.
+
+    Returns
+    -------
+    times : array, shape (n_times,)
+        Time points for signal plotting.
+    ampmat : ndarray (nfreq, n_samples)
+        Amplitude traces.
+    traces : ndarray, shape (n_epochs, n_times)
+        Normalized amplitude traces.
+    avg : array, shape (n_times,)
+        Average (or evoked) signal.
+    trigger_inds : array (trigger_inds,)
+        Indices of maxima in the phase signal.
+
+    References
+    ----------
+    [1] High gamma power is phase-locked to theta oscillations in human neocortex.
+    Canolty RT1, Edwards E, Dalal SS, Soltani M, Nagarajan SS, Kirsch HE,
+    Berger MS, Barbaro NM, Knight RT.
+    (Science. 2006)
+    """
+
+    if data.ndim != 1:
+        raise ValueError('Dimensions of data incorrect. Please use 1d array.')
+
+    Ws = morlet(sfreq, [phase_freq], n_cycles=n_cycles,
+                sigma=None, zero_mean=True)
+    data = data.reshape(1, data.size)
+    x_low = cwt(data, Ws, use_fft=False).ravel()
+
+    phases = np.angle(x_low)
+    trigger_inds, _ = peak_finder(phases)
+
+    n_samples = data.size
+    n_freqs = len(freqs)
+
+    if tmin >= tmax:
```

Comment:
So to my understanding, tmin and tmax define the window around the `event`, which in this case is the peak of the low-frequency wave, right? So tmin should always be < 0 (aka, to the left of the peak) and tmax always > 0 (aka to the right of the peak), right?